django_adobesign/backend.py

```diff
@@ -64,20 +64,19 @@ def create_signature(self, signature, post_sign_redirect_url=None,
         self.map_adobe_signer_to_signer(signature)
         return signature
 
-    def update_signer_status(self, signer, status):
-        raise NotImplementedError()
-
     def map_adobe_signer_to_signer(self, signature):
-        # Can raise a Signer.DoesNotExist
+        signers = signature.signers.all()
         for adobe_signer in self.get_all_signers(
                 signature.signature_backend_id).get('participantSets', []):
-            # We only have 1 signer by turn
-            email = adobe_signer['memberInfos'][0]['email'].lower()
-            signer = signature.signers.get(signing_order=adobe_signer['order'],
-                                           email=email)
+            # retrieve the right adobe signer by email and order values
+            # Can raise a StopIteration if signer not exist
+            signer = next(
+                (signer for signer in signers
+                 if signer.email == adobe_signer[
+                     'memberInfos'][0]['email'].lower() and
+                 signer.signing_order == adobe_signer['order']), None)
             signer.signature_backend_id = adobe_signer['id']
```

Comment:
(signer can be None)

Reply:
No it cannot... we made a `get` query, this can be raise a  `Signer.DoesNotExist` but it should not, we decided to let him raise a `Signer.DoesNotExist`...

Reply:
then why do we have the `None` as default in the `next(..., None)` ?

Reply:
Ok an error you right